tests/functional/src/main/java/org/mule/functional/junit4/FunctionalTestCase.java

```diff
@@ -315,6 +315,33 @@ public MuleEvent process() throws Exception
     }
 
     /**
+     * Runs the given flow asynchronously with a default event inside a transaction
+     * 
+     * @param flowName the name of the flow to be executed
+     * @param payload the payload to use in the message
+     * @param action See {@link TransactionConfig} constants
+     * @param factory See {@link MuleTransactionConfig#setFactory(TransactionFactory)}.
+     * @throws Exception
+     */
+    protected <T> void runTransactionalFlowAsync(String flowName, T payload, byte action, TransactionFactory factory) throws Exception
```

Comment:
There are many of this runFlow and runAsyncFlow methods. Can you create another jira so we can spend some time finding a more clean way to do the same thing?